### PR TITLE
DAOS-1338 Add destroy-tx (tx-abort) to the API, DAOS-1339, DAOS-3132 …

### PR DESCRIPTION
**Pull request by dinghwah**
…and update security init test with full_regression

Updated file: src/utils/py/daos_api.py
Updated file: src/tests/ftest/security/pool_connect_init.py
Updated file: src/tests/ftest/object/object_integrity.yaml
Updated file: src/tests/ftest/object/object_integrity.py

Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

### CODE REVIEW

File: src/tests/ftest/security/pool_connect_init.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def test_poolconnect(self):
                                       (permission denied with error -1001)
            Above 3 testcases are defined in the yaml file.
 
-        :avocado: tags=all,pr,medium,pool,security_basic,security
+        :avocado: tags=all,pr,full_regression,small,pool,sec_basic,security
```
Comment (saurabhtandan) [inline src/tests/ftest/security/pool_connect_init.py]: I think the idea is to either have a test in 'pr' group or 'full_regression' group. Not in both.

File: src/tests/ftest/object/object_integrity.py [hunk 2]
```diff
@@ -135,10 +135,12 @@ def test_invalid_tx_commit_close(self):
             (1)DAOS-1346: Verify commit tx bad parameter behavior.
             (2)DAOS-1343: Verify tx_close bad parameter behavior.
             (3)DAOS-1342: Verify tx_close through daos_api.
+            (4)DAOS-1338: Add and verify tx_abort through daos_api.
+            (5)DAOS-1339: Verify tx_abort bad parameter behavior.
         Test Description:
             Write Avocado Test to verify commit tx and close tx
                           bad parameter behavior.
-        :avocado: tags=all,pr,small,negative_test,neg_tx_commit,medium,vm
+        :avocado: tags=all,object,full_regression,small,single_object
```
Comment (saurabhtandan) [inline src/tests/ftest/object/object_integrity.py]: unique test tag is missing.

File: src/tests/ftest/object/object_integrity.py [hunk 3]
```diff
@@ -192,14 +194,56 @@ def test_invalid_tx_commit_close(self):
...
+
 
     @avocado.fail_on(DaosApiError)
     def test_single_object_validation(self):
         """
         Test ID: DAOS-707
         Test Description: Write Avocado Test to verify single data after
                           pool/container disconnect/reconnect.
-        :avocado: tags=all,object,full_regression,small,single_object,
+        :avocado: tags=all,object,full_regression,small,single_object
```
Comment (saurabhtandan) [inline src/tests/ftest/object/object_integrity.py]: unique test tag is missing.

File: src/tests/ftest/security/pool_connect_init.py (hunk 1 above)
Comment (dinghwah) [reply src/tests/ftest/security/pool_connect_init.py]: Thank you for the review and comment. 
It was requested by DE and System test to add this testcase to regression, I would keep both tags.

File: src/tests/ftest/object/object_integrity.py (hunk 3 above)
Comment (dinghwah) [reply src/tests/ftest/object/object_integrity.py]: Thanks for the comment.
Please see the new commit for the fix.

File: src/tests/ftest/object/object_integrity.py (hunk 2 above)
Comment (dinghwah) [reply src/tests/ftest/object/object_integrity.py]: same as above...
Please see the fix in the new commit.

**Pull request merged**